sfge/src/test/java/com/salesforce/graph/ops/ApexPathUtilLoopsTest.java

```diff
@@ -0,0 +1,189 @@
+
+        ApexStringValue value = visitor.getSingletonResult();
+        MatcherAssert.assertThat(value.isIndeterminant(), equalTo(true));
+    }
+
+    @Test
+    public void testForEachLoopWithNonForLoopMethodCall() {
+        String sourceCode =
+            "public class MyClass {\n"
+                + "   public void doSomething() {\n"
+                + "       String[] myList = new String[]{'hi','hello'};\n"
+                + "       for (String myString: myList) {\n"
+                + "           debug1(100);\n"
+                + "       }\n"
+                + "   }\n"
+                + "   public void debug1(Integer int) {\n"
+                + "       System.debug(int);\n"
+                + "   }\n"
+                + "}\n";
+
+        TestRunner.Result<SystemDebugAccumulator> result = TestRunner.get(g, sourceCode).walkPath();
+        SystemDebugAccumulator visitor = result.getVisitor();
+
+        ApexIntegerValue value = visitor.getSingletonResult();
+        MatcherAssert.assertThat(value.getValue().get(), equalTo(100));
+    }
+
+    @Test
+    public void testForLoopWithNonForLoopMethodCall() {
+        String sourceCode =
+            "public class MyClass {\n"
+                + "   public void doSomething() {\n"
+                + "       String[] myList = new String[]{'hi','hello'};\n"
+                + "       for (Integer i = 0; i < myList.size(); i++) {\n"
+                + "           debug1(100);\n"
+                + "       }\n"
+                + "   }\n"
+                + "   public void debug1(Integer int) {\n"
+                + "       System.debug(int);\n"
+                + "   }\n"
+                + "}\n";
+
+        TestRunner.Result<SystemDebugAccumulator> result = TestRunner.get(g, sourceCode).walkPath();
+        SystemDebugAccumulator visitor = result.getVisitor();
+
+        ApexIntegerValue value = visitor.getSingletonResult();
+        MatcherAssert.assertThat(value.getValue().get(), equalTo(100));
+    }
+
+    @Test
+    public void testWhileLoopWithMethodCall() {
+        String sourceCode =
+            "public class MyClass {\n"
+                + "   public void doSomething() {\n"
+                + "       String[] myList = new String[]{'hi','hello'};\n" +
+                "           Integer i = 0;\n"
+                + "       while (i < myList.size()) {\n"
+                + "           debug1(myList[i]);\n" +
+                "           i++;\n"
+                + "       }\n"
+                + "   }\n"
+                + "   public void debug1(String str) {\n"
+                + "       System.debug(str);\n"
+                + "   }\n"
+                + "}\n";
+
+        TestRunner.Result<SystemDebugAccumulator> result = TestRunner.get(g, sourceCode).walkPath();
+        SystemDebugAccumulator visitor = result.getVisitor();
+
+        ApexStringValue value = visitor.getSingletonResult();
+        MatcherAssert.assertThat(value.getValue().get(), equalTo("hi"));
```

Comment:
That's a good catch. I was using this test to study the behavior of while-loops. Ideally, this should've been a looped apex value with information on both "hi" and "hello". Let me modify this test to check for the correct thing and fail so that I can disable it.